File: src/business-services/src/main/java/com/itechart/security/business/model/dto/ContactDto.java
package com.itechart.security.business.model.dto;

import lombok.Getter;
import lombok.Setter;

import java.util.Date;
import java.util.Set;

@Getter
@Setter
public class ContactDto {
    private Long id;

    private String firstNameEn;

    private String lastNameEn;

    private String firstName;

    private String lastName;

    private String patronymic;

    private Date dateOfBirth;

    private Boolean isMale;

    private String nationality;

    private String photoUrl;

    private String industry;

    private String position;

    private Long linkedInId;

    private Set<OrderDto> orders;

    private Set<MessengerAccountDto> messengers;

    private Set<SocialNetworkAccountDto> socialNetworks;

    private Set<TelephoneDto> telephones;

    private Set<AddressDto> addresses;

    private Set<WorkplaceDto> workplaces;

    private Set<EmailDto> emails;

    private Set<AttachmentDto> attachments;

    private Set<SkillDto> skills;

    private HistoryEntryDto history;

    private Set<UniversityEducationDto> educations;

    private Set<ContactCommentDto> comments;

    private Set<LanguageDto> languages;

    private Long userId;

    private Date readingCommentsDate;

    private Integer countUnreadComments;
}

File: src/app/shared/models/contacts/ContactDto.ts
import {OrderDto} from "./OrderDto";
import {GridEntity} from "../grid/grid-entity";

export class ContactModel extends GridEntity {
  id: number = 0;
  firstNameEn: string = '';
  lastNameEn: string = '';
  firstName: string = '';
  lastName: string = '';
  patronymic: string = '';
  dateOfBirth: string = '';
  isMale: boolean;
  nationality: string = '';
  photoUrl: string;
  industry: string;
  position: string;
  linkedInId: number;
  orders: OrderDto[] = [];
  MessengerAccountDto: any[] = [];
  socialNetworks: any[] = [];
  telephones: any[] = [];
  addresses: any[] = [];
  workplaces: any[] = [];
  emails: any[] = [];
  attachments: any[] = [];
  skills: any[] = [];
  history: any = null;
  educations: any[] = [];
  comments: any[] = [];
  languages: any[] = [];
  userId: number;
  readingCommentsDate: string;
  countUnreadComments: number;

  selected: false;

  /*Grid fields*/
  fullName: string;
  firstEmail: string;
  firstTelephone: string;
  firstAddress: string;
}
